Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm

```diff
@@ -3256,6 +3256,25 @@ - (void)_loadAndDecodeImage:(NSURLRequest *)request constrainedToSize:(CGSize)ma
     });
 }
 
+- (void)_getInformationFromImageData:(NSData *)imageData completionHandler:(void (^)(BOOL canBeRepresentedAsSVG, NSArray<NSValue *> *availableSizes, NSError *error))completionHandler
+{
+    _page->getInformationFromImageData(makeVector(imageData), [completionHandler = makeBlockPtr(completionHandler)](bool canBeRepresentedAsSVG, const std::optional<Vector<WebCore::FloatSize>>& sizes) mutable {
+        if (!sizes) {
+            NSError *error = [NSError errorWithDomain:WKErrorDomain code:WKErrorUnknown userInfo:@{ NSLocalizedDescriptionKey: @"Failed to decode data" }];
+            return completionHandler(canBeRepresentedAsSVG, nil, error);
+        }
+
+        RetainPtr availableSizes = createNSArray(*sizes, [](auto& size) {
+#if PLATFORM(MAC)
+            return [NSValue valueWithSize:size];
+#else
+            return [NSValue valueWithCGSize:(CGSize)size];
```

Comment:
Is there a cleaner way to convert to `CGSize` instead of a C style cast? Maybe just a local variable?

Reply:
Updated to use local variable.